R/JHU.R

```diff
@@ -0,0 +1,151 @@
+    ),
+    # nolint end
+    #' @field source_data_cols existing columns within the raw data
+    source_data_cols = c("confirmed", "deaths", "recovered"),
+
+    #' @description Set up a table of region codes for clean data
+    #' @importFrom tibble tibble
+    set_region_codes = function() {
+      self$codes_lookup$`1` <- JHU_codes
+    },
+
+    #' @description Google specific download, calls `DataClass` download but
+    #' also fills the country info field
+    download = function() {
+      super$download()
+      if (!(is.na(self$supported_region_names[["1"]]))) {
+        self$country_info <- unique(
+          self$data$raw$daily_confirmed[["Country/Region"]]
+        )
+      }
+    },
+
+    #' @description JHU specific data cleaning. Joins the raw data sets, checks
+    #' column types and renames where needed.
+    #' @importFrom dplyr last_col bind_rows mutate rename select everything
+    #' @importFrom tidyr pivot_longer pivot_wider replace_na
+    #' @importFrom lubridate mdy
+    #' @importFrom rlang .data
+    clean_common = function() {
+      self$data$clean <- lapply(self$data$raw,
+        pivot_longer,
+        cols = 5:last_col(),
+        names_to = "Date",
+        values_to = "value"
+      )
+      self$data$clean <- self$data$clean %>%
+        bind_rows(.id = "variable") %>%
+        pivot_wider(names_from = variable, values_from = value) %>%
+        mutate(
+          Date = mdy(.data$Date),
+          daily_confirmed = as.numeric(.data$daily_confirmed),
+          daily_deaths = as.numeric(.data$daily_deaths),
+          daily_recovered = as.numeric(.data$daily_recovered),
+          level_2_region_code = NA_real_
+        ) %>%
+        rename(
+          date = .data$Date,
+          level_2_region = .data$`Province/State`,
+          level_1_region = .data$`Country/Region`,
+          cases_total = .data$daily_confirmed,
+          deaths_total = .data$daily_deaths,
+          recovered_total = .data$daily_recovered
+        ) %>%
+        replace_na(
+          list(
+            level_1_region = "Unknown",
+            level_2_region = "Unknown"
+          )
+        ) %>%
+        left_join(
+          self$codes_lookup$`1`,
+          by = c("level_1_region" = "Country_Region")
+        ) %>%
+        select(
+          date,
+          level_1_region,
+          level_1_region_code,
+          level_2_region,
+          level_2_region_code,
+          cases_total,
+          deaths_total,
+          recovered_total,
+          everything()
+        )
+    },
+
+    #' @description JHU specific country level data cleaning. Aggregates the
+    #' data to the country (level 2) level.
+    #' @importFrom dplyr select summarise group_by across everything
+    #' @importFrom rlang .data
+    clean_level_1 = function() {
+      self$data$clean <- self$data$clean %>%
+        group_by(
+          .data$date, .data$level_1_region_code,
```

Comment:
this is exactly the same as in the google class? Again consider adding to a parent class vs copying code without reason.,